src/program/packetblaster/packetblaster.lua

```diff
@@ -23,40 +23,51 @@ local function is_device_suitable (pcidev, patterns)
    end
 end
 
-function run_loadgen (c, patterns, duration)
-  local nics = 0
-  pci.scan_devices()
-  for _,device in ipairs(pci.devices) do
-    if is_device_suitable(device, patterns) then
-      nics = nics + 1
-      local name = "nic"..nics
-      config.app(c, name, LoadGen, device.pciaddress)
-      config.link(c, "source."..tostring(nics).."->"..name..".input")
-    end
-  end
-  assert(nics > 0, "<PCI> matches no suitable devices.")
-  engine.busywait = true
-  engine.configure(c)
-  local fn = function ()
-                print("Transmissions (last 1 sec):")
-                engine.report_apps()
-             end
-  local t = timer.new("report", fn, 1e9, 'repeating')
-  timer.activate(t)
-  if duration then engine.main({duration=duration})
-  else             engine.main() end
+function run_loadgen (c, patterns, opts)
+   assert(type(opts) == "table")
+   local nics = 0
+   pci.scan_devices()
+   for _,device in ipairs(pci.devices) do
+      if is_device_suitable(device, patterns) then
+         nics = nics + 1
+         local name = "nic"..nics
+         if opts.loop then
+            config.app(c, name, LoadGen, device.pciaddress)
+         else
+            config.app(c, name, Intel82599, {pciaddr = device.pciaddress})
```

Comment:
What I noticed is that Loadgen replays packets continuously, that's why I needed to make this difference. For instance, consider the following script that prints out received packets on a NIC:

```lua
module(..., package.seeall)

local Intel82599 = require("apps.intel.intel_app").Intel82599
local basic_apps = require("apps.basic.basic_apps")

function run (args)
   local pciaddr = assert(args[1], "No NIC")
   
   local c = config.new()
   config.app(c, "nic", Intel82599, {
      pciaddr = pciaddr,
   }) 
   config.app(c, "sink", basic_apps.Sink)
   config.link(c, "nic.tx -> sink.in1")

   engine.configure(c)
   engine.main({duration=5, report = { showlinks = true }})
end
```

If I always use LoadGen:

```
$ sudo ./snabb packetblaster replay --no-loop v4v6-256.pcap 81:00.1
filename=v4v6-256.pcap
link report:
    256 sent on pcap.output -> source.input (loss rate: 0%)
    256 sent on source.1 -> nic1.input (loss rate: 0%)

$ sudo ./snabb receiver 81:00.0                                                                                                                       
link report:
    1,385,473 sent on nic.tx -> sink.in1 (loss rate: 0%)
```

If I use Intel82599 when no-loop:

```
$ sudo ./snabb packetblaster replay --no-loop v4v6-256.pcap 81:00.1
filename=v4v6-256.pcap
link report:
    256 sent on pcap.output -> source.input (loss rate: 0%)
    256 sent on source.1 -> nic1.rx (loss rate: 0%)

$ sudo ./snabb receiver 81:00.0                                                                                                                       
link report:
    256 sent on nic.tx -> sink.in1 (loss rate: 0%)
```

OTOH, I realized there's a bug and the link should be "nic.rx" for the Intel82599 case.